lib/media/track/remotevideotrack.js

```diff
@@ -125,8 +134,14 @@ class RemoteVideoTrack extends RemoteMediaVideoTrack {
     const result = super.detach(el);
     const elements = Array.isArray(result) ? result : [result];
     elements.forEach(element => {
-      this._intersectionObserver.unobserve(element);
-      this._resizeObserver.unobserve(element);
+      if (this._idleTrackSwitchOff) {
+        this._intersectionObserver.unobserve(element);
+      }
+
+      if (this._renderHints) {
+        this._resizeObserver.unobserve(element);
+      }
+
```

Comment:
I don't think we need the `if()` check since you are using the `NullObserver`.

Reply:
You are right that this check is not necessary, but I it helps with unit tests, as they can verify that unobserve being called only when option is set.

Reply:
@makarandp0 I think the unit tests can test if the correct callback (actual or fake) is called. That way, you can remove these checks.